### DISCUSSION TITLE
Deterministic module IDs are unstable after I update any devDependencies

### DISCUSSION THREAD

**Discussion by stanleyxu2005** [webpack/webpack#18046] [General]
It's strange that after I update any of my devDepencencies (no change in source code), all built js files will be slightly changed. (see fig. 1)

![image](https://github.com/webpack/webpack/assets/5162687/0e3f0f7f-1054-449f-8b21-28f880862192)

It is 100% reproducible. It seems that Webpack calculates moduleId with some modules from a dev dependency? 
I'm afraid that I've configured something wrong in Webpack. 

Here I attached the part of  `splitChunk`, hope someone can shed some lights.
![image](https://github.com/webpack/webpack/assets/5162687/b60f9896-5ba3-4857-bacb-22f5e0330cb4)

Also attached the full devDepencendies and dependencies
![image](https://github.com/webpack/webpack/assets/5162687/a49bb000-bfd3-457c-84f6-92bfa8b789ca)

Additional information can be found in [the comment of issue 14521](https://github.com/webpack/webpack/issues/14521#issuecomment-1924161957)

Cheers

**Comment by alexander-akait** [comment]
Please read https://github.com/webpack/webpack/issues/14521#issuecomment-1924173813
